src/false_positives.h

```diff
@@ -0,0 +1,115 @@
+    {0, 62, 123, 184, 244, 304, 363, 422, 481, 540, 599, 657, 716, 774, 832, 890, 948, 1006, 1064, 1122, 1179, 1237, 1295, 1352, 1409, 1467, 1524, 1581, 1639, 1696, 1753, 1810, 1867, 1924, 1981, 2038, 2094, 2151, 2208, 2265, 2321, 2378, 2435, 2491, 2548, 2604, 2661, 2717, 2774, 2830, 2886, 2943, 2999, 3055, 3111, 3168, 3224, 3280, 3336, 3392, 3448, 3504, 3560, 3617, 3673, 3728, 3784, 3840, 3896, 3952, 4008, 4064, 4120, 4176, 4231, 4287, 4343, 4399, 4454, 4510, 4566, 4621, 4677, 4733, 4788, 4844, 4899, 4955, 5010, 5066, 5122, 5177, 5232, 5288, 5343, 5399, 5454, 5510, 5565, 5620, 5676, 5731, 5786, 5842, 5897, 5952, 6008, 6063, 6118, 6173, 6229, 6284, 6339, 6394, 6449, 6504, 6560, 6615, 6670, 6725, 6780, 6835, 6890, 6945, 7000, 7055, 7110, 7165, 7220, 7275},
+    {0, 63, 125, 187, 248, 309, 369, 429, 489, 549, 609, 668, 728, 787, 846, 905, 964, 1023, 1082, 1141, 1199, 1258, 1317, 1375, 1433, 1492, 1550, 1608, 1667, 1725, 1783, 1841, 1899, 1957, 2015, 2073, 2130, 2188, 2246, 2304, 2361, 2419, 2477, 2534, 2592, 2649, 2707, 2764, 2822, 2879, 2936, 2994, 3051, 3108, 3165, 3223, 3280, 3337, 3394, 3451, 3508, 3565, 3622, 3680, 3737, 3793, 3850, 3907, 3964, 4021, 4078, 4135, 4192, 4249, 4305, 4362, 4419, 4476, 4532, 4589, 4646, 4702, 4759, 4816, 4872, 4929, 4985, 5042, 5098, 5155, 5212, 5268, 5324, 5381, 5437, 5494, 5550, 5607, 5663, 5719, 5776, 5832, 5888, 5945, 6001, 6057, 6114, 6170, 6226, 6282, 6339, 6395, 6451, 6507, 6563, 6619, 6676, 6732, 6788, 6844, 6900, 6956, 7012, 7068, 7124, 7180, 7236, 7292, 7348, 7404},
+    {0, 64, 127, 190, 252, 314, 375, 436, 497, 558, 619, 679, 740, 800, 860, 920, 980, 1040, 1100, 1160, 1219, 1279, 1339, 1398, 1457, 1517, 1576, 1635, 1695, 1754, 1813, 1872, 1931, 1990, 2049, 2108, 2166, 2225, 2284, 2343, 2401, 2460, 2519, 2577, 2636, 2694, 2753, 2811, 2870, 2928, 2986, 3045, 3103, 3161, 3219, 3278, 3336, 3394, 3452, 3510, 3568, 3626, 3684, 3743, 3801, 3858, 3916, 3974, 4032, 4090, 4148, 4206, 4264, 4322, 4379, 4437, 4495, 4553, 4610, 4668, 4726, 4783, 4841, 4899, 4956, 5014, 5071, 5129, 5186, 5244, 5302, 5359, 5416, 5474, 5531, 5589, 5646, 5704, 5761, 5818, 5876, 5933, 5990, 6048, 6105, 6162, 6220, 6277, 6334, 6391, 6449, 6506, 6563, 6620, 6677, 6734, 6792, 6849, 6906, 6963, 7020, 7077, 7134, 7191, 7248, 7305, 7362, 7419, 7476, 7533},
+};
+
+size_t compute_capacity(uint32_t bits, size_t max_elements, uint32_t fpbits) {
+    if (max_elements > MAX_ELEM) return max_elements;
```

Comment:
That's a little footgunny? Capacities of 129 aren't really that absurd. At a minimum wouldn't it be better to return max_elements + ceil(fpbits/bits) as a conservative approximation?   It looks to me that at least for larger bit counts the continuations of your tables past the maximum number of elements are almost exactly fit by a linear model (essentially up to rounding).

E.g. 14-bits last 20 elements fit a linear model +/- 0.47,  64 bits last 20 elements fit a linear model +/- 0.52.

Reply:
This isn't clear from the code at all, but that's actually unnecessary.

For capacities over 129, the probability for a false positive is always below 2^-256, so just using capacity = max_elements is always ok (the API documentation doesn't guarantee anything for probabilities below 2^-256).

There is an alternative table representation that makes that clearer. I'll push it to a branch soon.

Reply:
oh, okay! I did the math and see it.   It should also, by the same argument, be the case that e.g. capacity 117 should always decode up to 128 elements with a low (1:2^17) false positive rate. (assuming the decoder knows its allowed to have a higher output degree) No?

Reply:
Updated to a very different approach. It should now deal with basically arbitrary field sizes exactly, and generally give okayish results for fpbits > 256.

I don't think it's true that it works the other direction. Even though there are fewer capacity-117 sketches with up to 128 elements than there are 117-capacity bit arrays, there is no guarantee that there are no collisions between them.